cache/network/download.go

```diff
@@ -36,7 +38,12 @@ func Download(params DownloadParams, logger log.Logger) (matchedKey string, err
 		return "", fmt.Errorf("cache key list is empty")
 	}
 
-	client := newAPIClient(retryhttp.NewClient(logger), params.APIBaseURL, params.Token, logger)
+	var client apiClient
+	if params.FeatureFlags[cache.Multithreading] == true {
```

Comment:
Feature flag handling should happen outside the cache package. The cache package should not know about feature flags -> it should expose the related parameters, but should depend on those input params not on feature flags.

Reasoning: adding feature flag handling makes the cache package more complex, breaks the single responsibility principle.